spec/features/admin/newsletter/emails_spec.rb

```diff
@@ -0,0 +1,131 @@
+        expect(page).to have_content newsletter.subject
+        expect(page).to have_content I18n.t("admin.newsletters.recipients.#{newsletter.segment_recipient}")
+      end
+    end
+  end
+
+  scenario "Create" do
+    visit admin_newsletters_path
+    click_link "New newsletter"
+
+    fill_in "newsletter_subject", with: "This is a subject"
+    select "Proposal authors", from: "newsletter_segment_recipient"
+    fill_in "newsletter_from", with: "noreply@consul.dev"
+    fill_in "newsletter_body", with: "This is a body"
+    click_button "Create Newsletter"
+
+    expect(page).to have_content "Newsletter created successfully"
+    expect(page).to have_content "This is a subject"
+    expect(page).to have_content "Proposal authors"
+    expect(page).to have_content "noreply@consul.dev"
+    expect(page).to have_content "This is a body"
+  end
+
+  scenario "Update" do
+    newsletter = create(:newsletter)
+
+    visit admin_newsletters_path
+    within("#newsletter_#{newsletter.id}") do
+      click_link "Edit"
+    end
+
+    fill_in "newsletter_subject", with: "This is a different subject"
+    select "Investment authors in the current budget", from: "newsletter_segment_recipient"
+    fill_in "newsletter_from", with: "no-reply@consul.dev"
+    fill_in "newsletter_body", with: "This is different a body"
+    click_button "Update Newsletter"
+
+    expect(page).to have_content "Newsletter updated successfully"
+    expect(page).to have_content "This is a different subject"
+    expect(page).to have_content "Investment authors in the current budget"
+    expect(page).to have_content "no-reply@consul.dev"
+    expect(page).to have_content "This is different a body"
+  end
+
+  scenario "Destroy" do
+    newsletter = create(:newsletter)
+
+    visit admin_newsletters_path
+    within("#newsletter_#{newsletter.id}") do
+      click_link "Delete"
+    end
+
+    expect(page).to have_content "Newsletter deleted successfully"
+    expect(page).to have_css(".newsletter", count: 0)
+  end
+
+  scenario 'Errors on create' do
+    visit new_admin_newsletter_path
+
+    click_button "Create Newsletter"
+
+    expect(page).to have_content error_message
+  end
+
+  scenario "Errors on update" do
+    newsletter = create(:newsletter)
+    visit edit_admin_newsletter_path(newsletter)
+
+    fill_in "newsletter_subject", with: ""
+    click_button "Update Newsletter"
+
+    expect(page).to have_content error_message
+  end
+
+  scenario "Send newsletter email", :js do
+    newsletter = create(:newsletter)
+    visit admin_newsletter_path(newsletter)
+
+    click_link "Send"
+
+    total_users = newsletter.list_of_recipients.count
+    page.accept_confirm("Are you sure you want to send this newsletter to #{total_users} users?")
+
+    expect(page).to have_content "Newsletter sent successfully"
+  end
+
+  scenario "Select list of users to send newsletter" do
+    Newsletter.segment_recipients.each do |user_group, n|
```

Comment:
If you're only going to use the keys of a Hash iterating with .each (as I see no use for `n` variable inside the loop) you can better use `each_key` like `.each_key do |user_group|` . It's also a rubocop performance cop [Performance/HashEachMethods](https://github.com/bbatsov/ruby-style-guide#hash-each) the one backing up this suggestion